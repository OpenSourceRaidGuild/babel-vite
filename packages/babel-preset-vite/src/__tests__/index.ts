import pluginTester from 'babel-plugin-tester'
import preset, { VitePresetOptions } from '..'

const fixture = (filename: string, options?: VitePresetOptions) => ({
  fixture: require.resolve(`./fixtures/${filename}`),
  babelOptions: {
    presets: [options ? [preset, options] : preset]
  }
})

pluginTester({
  plugin: () => ({}),
  pluginName: 'vite',
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: {
    defaults: fixture('input'),
    all: fixture('input', { env: true, glob: true }),
    'env-only': fixture('input', { env: true, glob: false }),
    'glob-only': fixture('input', { env: false, glob: true }),
    'custom-env-config': fixture('input', { env: { envPrefix: 'TEST_PREFIX_' }, glob: true })
  }
})
