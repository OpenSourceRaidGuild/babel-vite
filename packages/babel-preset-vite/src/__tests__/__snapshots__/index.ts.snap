// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vite all: all 1`] = `

export const envVar = import.meta.env.VITE_VAR

export const modules = import.meta.glob('files/*.ts')

export const eagerModules = import.meta.globEager('files/*.ts')

export const hot = import.meta.hot

      ↓ ↓ ↓ ↓ ↓ ↓

export const envVar = process.env.VITE_VAR
export const modules = {
  './files/file1.ts': () => import('./files/file1.ts')
}
export const eagerModules = {
  './files/file1.ts': require('./files/file1.ts')
}
export const hot = module.hot


`;

exports[`vite defaults: defaults 1`] = `

export const envVar = import.meta.env.VITE_VAR

export const modules = import.meta.glob('files/*.ts')

export const eagerModules = import.meta.globEager('files/*.ts')

export const hot = import.meta.hot

      ↓ ↓ ↓ ↓ ↓ ↓

export const envVar = process.env.VITE_VAR
export const modules = {
  './files/file1.ts': () => import('./files/file1.ts')
}
export const eagerModules = {
  './files/file1.ts': require('./files/file1.ts')
}
export const hot = module.hot


`;

exports[`vite env-only: env-only 1`] = `

export const envVar = import.meta.env.VITE_VAR

export const modules = import.meta.glob('files/*.ts')

export const eagerModules = import.meta.globEager('files/*.ts')

export const hot = import.meta.hot

      ↓ ↓ ↓ ↓ ↓ ↓

export const envVar = process.env.VITE_VAR
export const modules = import.meta.glob('files/*.ts')
export const eagerModules = import.meta.globEager('files/*.ts')
export const hot = import.meta.hot


`;

exports[`vite glob-only: glob-only 1`] = `

export const envVar = import.meta.env.VITE_VAR

export const modules = import.meta.glob('files/*.ts')

export const eagerModules = import.meta.globEager('files/*.ts')

export const hot = import.meta.hot

      ↓ ↓ ↓ ↓ ↓ ↓

export const envVar = import.meta.env.VITE_VAR
export const modules = {
  './files/file1.ts': () => import('./files/file1.ts')
}
export const eagerModules = {
  './files/file1.ts': require('./files/file1.ts')
}
export const hot = import.meta.hot


`;

exports[`vite hot-only: hot-only 1`] = `

export const envVar = import.meta.env.VITE_VAR

export const modules = import.meta.glob('files/*.ts')

export const eagerModules = import.meta.globEager('files/*.ts')

export const hot = import.meta.hot

      ↓ ↓ ↓ ↓ ↓ ↓

export const envVar = import.meta.env.VITE_VAR
export const modules = import.meta.glob('files/*.ts')
export const eagerModules = import.meta.globEager('files/*.ts')
export const hot = module.hot


`;
